fix(transformer/styled-components): template literal expressions order is wrong after minification

Dunqing opened the pull request:
Related: https://github.com/oxc-project/oxc/pull/12202

`swap_remove` will swap the last element to the index you want to remove, so this will cause the expression's order to be incorrect. Here is order sensitive, so we can't use `swap_remove`.

Dunqing commented on crates/oxc_transformer/src/plugins/styled_components.rs:
```diff
@@ -404,28 +404,30 @@ impl<'a> StyledComponents<'a, '_> {
+                expressions.take_in(ctx.ast.allocator).into_iter().enumerate().filter_map(
+                    |(i, expr)| remained_expression_indices.contains(&i).then_some(expr),
+                ),
+            );
```
Thanks!

overlookmotel commented:
### Merge activity

* **Jul 11, 9:10 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Jul 11, 9:10 AM UTC**: `overlookmotel` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Dunqing committed:
fix(transformer/styled-components): template literal expressions order is wrong after minification (#12211)

Related: https://github.com/oxc-project/oxc/pull/12202

`swap_remove` will swap the last element to the index you want to remove, so this will cause the expression's order to be incorrect. Here is order sensitive, so we can't use `swap_remove`.
